### PR TITLE
Fixes Bug in Regular Sequence

### PR DESCRIPTION
**Pull request by derekagorhom**
fixes issue #7199 

fixed the bug in `Regular sequence` by changing the code in `line 182` 
to 
`ucrNudRepeatValues.AddAdditionalCodeParameterPair(clsSeqDateFunction, New RParameter("each", 1), iAdditionalPairNo:=1)`

@africanmathsinitiative/developers  can anyone review this.

### CODE REVIEW

**Comment by derekagorhom** [comment]
@Patowhiz here is the pull request

**Review by rdstern — DISMISSED** [review]
@derekagorhom and @Patowhiz it now seems fine to me.  I assume @Patowhiz has also checked whether anything need be changed for any other dialogues.
I am approving and very happy to see this correction.

**Comment by dannyparsons** [comment]
The `each` parameter should be there, but it is in `rep` not `seq`. The code examples that you shared were already correct, `each` was in `rep`.

**Comment by Patowhiz** [comment]
@dannyparsons you may have missed my `seq.Date` example, have a look at it below. The `each` is in the `seq.Date` and `rep`. So what I mean is, if @derekagorhom chooses to still use the `seq.Date`, then the `each` parameter passed to it should be removed. If he chooses to use the `seq` same change will automatically apply.

```
regular <- rep(x=seq.Date(from=as.Date(x="2018/1/1"), each=1, to=as.Date(x="2018/12/31"), by="1 days"), each=1, length.out=36)
data_book$add_columns_to_data(data_name="survey", col_name="regular", col_data=regular, before=FALSE)
rm(regular)
```

**Comment by derekagorhom** [comment]
> @dannyparsons you may have missed my `seq.Date` example, have a look at it below. The `each` is in the `seq.Date` and `rep`. So what I mean is, if @derekagorhom chooses to still use the `seq.Date`, then the `each` parameter passed to it should be removed. If he chooses to use the `seq` same change will automatically apply.
> 
> ```
> regular <- rep(x=seq.Date(from=as.Date(x="2018/1/1"), each=1, to=as.Date(x="2018/12/31"), by="1 days"), each=1, length.out=36)
> data_book$add_columns_to_data(data_name="survey", col_name="regular", col_data=regular, before=FALSE)
> rm(regular)
> ```

Thanks for reminding me @Patowhiz i will make the necessary changes

**Comment by derekagorhom** [comment]
@dannyparsons @Patowhiz after i commented out `clsRepFunction.AddParameter("each", 1, iPosition:=2)
` in` line 140`
i run the dates sub of the Regular on the script and i got this 

```
regular <- rep(x=seq.Date(from=as.Date(x="2018/1/1"), to=as.Date(x="2018/12/31"), by="1 days"), length.out=15527)
data_book$add_columns_to_data(data_name="mydata", col_name="regular", col_data=regular, before=FALSE)

rm(regular)
```

which is what I and @Patowhiz  wanted you to be aware of since `each` is not necessary for the dates dialogue.

**Comment by dannyparsons** [comment]
Thanks, yes sorry for the confusion. `each` is in `rep` only, that's correct.

**Comment by derekagorhom** [comment]
> @derekagorhom is this PR ready for test/review again or do you still have more changes to make? thanks

no please

File: instat/dlgRegularSequence.vb [hunk 1]
```diff
@@ -137,7 +137,7 @@ Public Class dlgRegularSequence
 
         clsRepFunction.SetRCommand("rep")
         clsRepFunction.AddParameter("x", clsRFunctionParameter:=clsSeqFunction, iPosition:=0)
-        clsRepFunction.AddParameter("each", 1, iPosition:=2)
+        'clsRepFunction.AddParameter("each", 1, iPosition:=2)
```
Comment (lloyddewit) [inline instat/dlgRegularSequence.vb:140]: delete this line instead of commenting it out?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by derekagorhom** [comment]
@lloyddewit  i have made the changes, please could you check again?

**Comment by Patowhiz** [comment]
@lloyddewit please note that this PR has been replaced by PR #7231

**Pull request closed without merging**